feat(perfmon): Polish up native performance monitoring (#4640)
getsentry/sentry-docs: src/platforms/common/performance/index.mdx

<!-- Include in `notSupported` any SDK which enables tracing automatically -->

<PlatformSection supported={["native"]}>

<Alert level="info" title="Important">

> This is currently an experimental feature. Performance monitoring is still a work-in-progress feature, and its API may be unstable. It may also have bugs. Feedback is appreciated, and should be submitted as an issue in the appropriate repository.

Reviewer: does it make sense to link directly to the sentry-native repo instead of being somewhat vague?

Author: i think it certainly makes sense, but i'm not entirely sure how or if the docs have the ability to do that right now.

a quick search through the repository for `packageData.repoUrl`  and `repoUrl` doesn't reveal too many jsx elements, notably none that simply acts as a smart link to the repository of the selected SDK. there's nothing stopping us from adding one in if it turns out this would actually be pretty useful (probably something @imatwawana would know best), but it would most likely belong in a different PR and i would take a little longer than usual to rig something like this up due to a lack of familiarity with this project's stack.

Reviewer 2: I could be wrong here, but I assumed the ask was just to add plain old link to here: https://github.com/getsentry/sentry-native/issues

Author: Sorry, let me clarify: I'm curious to know if something similar to `<PlatformIdentifier>` but instead provides automatically figures out what link to use to go to an SDK's repository would be useful in the docs in general. Are there any other places where you (or others) may have wanted to link to the repository inline with the text, ie "hey just go to the repo here `<AutoLinkToWhateverRepoYouNeedToGoToBasedOnTheSDKYoureOn>`"?

Reviewer 2: I don't know how often we'd use something like that, TBH. There's a link on the right side of every platform page that links you to the repo, so I don't know that we need another component specifically for inline text.

Author: In that case at the risk of making this note completely misleading, I'm going to just directly say "to the right" since that's where the link sites relative to it.

> <Alert level="info" title="Important">
> 
> This is currently an experimental feature. Performance monitoring is still a work-in-progress feature, and its API may be unstable. It may also have bugs. Feedback is appreciated, and should be submitted as an issue in the appropriate repository.
> 
> </Alert>

Reviewer 2: an early adopter/experimental feature is just a regular level note. also seems a bit redundant. i'm not married to the wording edit, feel free to change it back if you feel like the other version is conveying something more.

```suggestion
<Note>

Performance monitoring is still an experimental, work-in-progress feature, and its API may be unstable. It may also have bugs. Feedback is appreciated, and should be submitted as an issue in the appropriate repository.

</Note>
```

Author: The edit certainly makes it less wordy so I've decided to add that in. 👍
